src/plugins/data_view_management/public/components/edit_index_pattern/edit_index_pattern.tsx

```diff
@@ -82,7 +82,7 @@ export const EditIndexPattern = withRouter(
     }, [savedObjectsManagement]);
 
     useEffect(() => {
-      if (allowedTypes.length === 0) {
+      if (allowedTypes.length === 0 || !indexPattern.isPersisted()) {
```

Comment:
the plan was to support all the functionalities on the ad hoc dataviews that are supported on normal ones, this includes editing. If something is not working correctly it should be considered a bug and we are happy to fix it.

Reply:
The problem is, that without this check you're getting errors once you try to edit the ad hoc data view in stack management:
![Bildschirmfoto 2022-08-23 um 10 28 19](https://user-images.githubusercontent.com/463851/186112103-ba2acfd9-3b72-446c-a787-3804a5fa20f0.png)

Reply:
@ppisljar 

https://user-images.githubusercontent.com/463851/186112646-91545740-f866-406a-a117-048b9e187617.mp4
I've removed this check in an a-la-carte dev deployment so you can test: 

https://kertal-pr-138283-introduce-hoc-data-views.kbndev.co

Reply:
It seems apart from console error this works correctly.
here is a PR to fix the console error: https://github.com/elastic/kibana/pull/139376

i would suggest to remove this check in this PR and allow editing the ad hoc dataview.